### PR TITLE
Fix `ssc run` to be on par with `ssc build`

### PR DESCRIPTION
**Pull request by chicoxyzzy**
This PR replaces part of #394 except the Android changes that we can figure out later

### CODE REVIEW

**Review by mribbons — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
do not merge yet

File: src/cli/cli.cc [hunk 1]
```diff
@@ -3879,14 +3912,48 @@ int main (const int argc, const char* argv[]) {
...
         }
       }
+
+      if (hostArg.size() == 0) {
+        hostArg = optionValue("build", option, "--host");
+        if (hostArg.size() > 0) {
+          devHost = hostArg;
+        } else {
+          if (isForIOS || isForAndroid) {
+            auto r = exec((!platform.win)
```
Comment (jwerle) [inline src/cli/cli.cc:3937]: nice

File: src/cli/cli.cc [hunk 2]
```diff
@@ -3372,7 +3391,7 @@ int main (const int argc, const char* argv[]) {
     // MacOS Code Signing
     // ---
     //
-    if (flagCodeSign && platform.mac) {
+    if (flagCodeSign && platform.mac && !flagBuildForIOS) {
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:3394]: @heapwolf please confirm we can skip this for iOS

File: src/cli/cli.cc [hunk 3]
```diff
@@ -3478,7 +3497,7 @@ int main (const int argc, const char* argv[]) {
     // MacOS Notarization
     // ---
     //
-    if (flagShouldNotarize && platform.mac) {
+    if (flagShouldNotarize && platform.mac && !flagBuildForIOS) {
```
Comment (chicoxyzzy) [inline src/cli/cli.cc:3500]: @heapwolf please confirm we can skip this for iOS

File: src/cli/cli.cc [hunk 4]
```diff
@@ -3446,7 +3465,7 @@ int main (const int argc, const char* argv[]) {
     // MacOS Packaging
     // ---
     //
-    if (flagShouldPackage && platform.mac) {
+    if (flagShouldPackage && platform.mac && !flagBuildForIOS) {
```
[the remark this answers is not in the document]
Comment (heapwolf) [reply src/cli/cli.cc:3468]: you can notarize an app for macOS that is not for iOS, if that is the question

**Review by heapwolf — APPROVED** [review]
I think this is ok is the condition is to allow notarization of apps on macOS that are not for iOS.

**Pull request merged**